flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/ExecutionAttemptMappingProvider.java

```diff
@@ -38,32 +38,34 @@
     private final List<ExecutionVertex> tasks;
 
     /** The cached mapping, which would only be updated on miss. */
-    private final LinkedHashMap<ExecutionAttemptID, ExecutionVertex> cachedTasksById;
+    private volatile Map<ExecutionAttemptID, ExecutionVertex> cachedTasksById;
```

Comment:
Do we need it to be `volatile`? I believe we don't really care about the read outside of the `synchronized` block. IMO it is just fine if it reads from an older copy. If the thread local map still has the entry, good for us, we will return it. If not, we enter the synchronized block and update the variable.

Or am I missing something?

Reply:
I have removed the volatile. Initially, I was afraid that it is possible to read the broken link from the non-volatile variable but as I remember it is not a case since publishing links should be atomic.